### PR TITLE
Follow-up of the user-defined pruner tutorial

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

There are several TODOs as a follow-up of the tutorial of the user-defined pruner added in #2140. In my understanding, the author of the original PR is busy, and #1619 is still labelled as `contirbution-welcome`.

## Description of the changes

Mainly following @crcrpar 's comment in  #2140.

### CODE REVIEW

**Review by crcrpar — APPROVED** [review]
thank you.

File: tutorial/20_recipes/006_user_defined_pruner.py [hunk 1]
```diff
@@ -16,24 +16,25 @@
 The :func:`~optuna.study.create_study` constructor takes, as an optional
 argument, a pruner inheriting from :class:`~optuna.pruners.BasePruner`. The
 pruner should implement the abstract method
-:meth:`~optuna.pruners.BaseSampler.prune`, which takes arguments for the
+:func:`~optuna.pruners.BaseSampler.prune`, which takes arguments for the
```
Comment (himkt) [inline tutorial/20_recipes/006_user_defined_pruner.py:19]: > :func:`~optuna.pruners.BaseSampler.prune`

Should :func:`~optuna.pruners.BaseSampler.prune` be replaced with `:func:`~optuna.pruners.BasePruner.prune`? Sorry if I misunderstand something.

File: tutorial/20_recipes/006_user_defined_pruner.py [hunk 2]
```diff
@@ -43,10 +44,10 @@
 implement a pruner which terminates a trial at a certain step if it is in last
 place compared to completed trials at the same step. We begin considering
 pruning after a "warmup" of 1 training step and 5 completed trials. For
-demonstration purposes, we print() a diagnostic message from prune() when it is
-about to return True (indicating pruning).
+demonstration purposes, we :func:`print` a diagnostic message from ``prune`` when
```
Comment (himkt) [inline tutorial/20_recipes/006_user_defined_pruner.py:47]: I want to know why ``prune`` here is not documented with an rst method syntax ":func:`~optuna.xxx.prune`".

File: tutorial/20_recipes/006_user_defined_pruner.py (hunk 1 above)
Comment (crcrpar) [reply tutorial/20_recipes/006_user_defined_pruner.py:19]: You are right, good catch.

File: tutorial/20_recipes/006_user_defined_pruner.py (hunk 2 above)
Comment (crcrpar) [reply tutorial/20_recipes/006_user_defined_pruner.py:47]: The prune refers to the one implemented in the `LastPlacePruner`.
Therefore I don't think :func:~optuna.xxx.prune is appropriate in the line.

File: tutorial/20_recipes/006_user_defined_pruner.py (hunk 1 above)
Comment (nzw0301) [reply tutorial/20_recipes/006_user_defined_pruner.py]: Thank @himkt for your careful check and @crcrpar for your double-check!

File: tutorial/20_recipes/006_user_defined_pruner.py (hunk 2 above)
Comment (nzw0301) [reply tutorial/20_recipes/006_user_defined_pruner.py:47]: I agree with @crcrpar; this is why it isn't documented.

**Review by himkt — APPROVED** [review]
I apologize that I missed the change. LGTM, thank you @nzw0301!

**Pull request merged**